app/services/family_finances/attendee_report.rb

```diff
@@ -18,12 +18,20 @@ def cost_reports
        conferences_cost, rec_center_cost]
     end
 
-    def on_campus_stays
-      stay_scope.select(&:dormitory?).map(&method(:stay_row))
+    def on_campus_stays(without_unit: false)
+      if without_unit
+        stay_scope.select(&:dormitory?).map(&method(:stay_row_without_unit))
+      else
+        stay_scope.select(&:dormitory?).map(&method(:stay_row))
+      end
```

Comment:
Again we could eliminate some redundancy here (and favour ternary statement over if statement to discourage branching logic).
```
row_method = without_unit ? :stay_row_without_unit : :stay_row
stay_scope.select(&:dormitory?).map(&method(row_method))
```